traffic_ops/app/t/api/1.2/tenant.t

```diff
@@ -0,0 +1,156 @@
+		})
+		->status_is(200)->or( sub { diag $t->tx->res->content->asset->{content}; } )
+		->json_is( "/response/name" => "tenantA2" )
+		->json_is( "/response/id" => $tenantA_id )
+		->json_is( "/response/active" => $true )
+		->json_is( "/response/parentId" => $root_tenant_id )
+		->json_is( "/alerts/0/level" => "success" )
+	, 'Does the tenantA2 details return?';
+
+#change "active"
+ok $t->put_ok('/api/1.2/tenants/' . $tenantA_id  => {Accept => 'application/json'} => json => {
+			"name" => "tenantA2", "active" => $false, "parentId" => $root_tenant_id 
+		})
+		->status_is(200)->or( sub { diag $t->tx->res->content->asset->{content}; } )
+		->json_is( "/response/name" => "tenantA2" )
+		->json_is( "/response/id" => $tenantA_id )
+		->json_is( "/response/active" => $false )
+		->json_is( "/response/parentId" => $root_tenant_id )
+		->json_is( "/alerts/0/level" => "success" )
+	, 'Did we moved to non active?';
+
+#change "active" back
+ok $t->put_ok('/api/1.2/tenants/' . $tenantA_id  => {Accept => 'application/json'} => json => {
+			"name" => "tenantA2", "active" => $true, "parentId" => $root_tenant_id 
+		})
+		->status_is(200)->or( sub { diag $t->tx->res->content->asset->{content}; } )
+		->json_is( "/response/name" => "tenantA2" )
+		->json_is( "/response/id" => $tenantA_id )
+		->json_is( "/response/active" => $true )
+		->json_is( "/response/parentId" => $root_tenant_id )
+		->json_is( "/alerts/0/level" => "success" )
+	, 'Did we moved back to active?';
+
+#cannot change tenant parent to undef
+ok $t->put_ok('/api/1.2/tenants/' . $tenantA_id  => {Accept => 'application/json'} => json => {
+			"name" => "tenantC", 
+		})->status_is(400);
+
+#cannot change root-tenant to inactive
+ok $t->put_ok('/api/1.2/tenants/' . $root_tenant_id  => {Accept => 'application/json'} => json => {
+			"name" => "root", "active" => $false, "parentId" => undef  
+		})->status_is(400);
+
+#adding a child tenant
+ok $t->post_ok('/api/1.2/tenants' => {Accept => 'application/json'} => json => {
+        "name" => "tenantD", "active" => $true, "parentId" => $tenantA_id })->status_is(200)->or( sub { diag $t->tx->res->content->asset->{content}; } )
+	->json_is( "/response/name" => "tenantD" )
+	->json_is( "/response/active" => $true )
+	->json_is( "/response/parentId" =>  $tenantA_id)
+            , 'Does the tenant details return?';
+
+#adding a child inactive tenant
+ok $t->post_ok('/api/1.2/tenants' => {Accept => 'application/json'} => json => {
+        "name" => "tenantE", "active" => $false, "parentId" => $tenantA_id })->status_is(200)->or( sub { diag $t->tx->res->content->asset->{content}; } )
+	->json_is( "/response/name" => "tenantE" )
+	->json_is( "/response/active" => $false )
+	->json_is( "/response/parentId" =>  $tenantA_id)
+            , 'Does the tenant details return?';
+
+#cannot delete a tenant that have children
+ok $t->delete_ok('/api/1.2/tenants/' . $tenantA_id)->status_is(500);
```

Comment:
how about returning a 400 instead in the code rather than letting the db throw a 500?